lib/vm/src/instance.rs

```diff
@@ -273,95 +259,15 @@ impl Instance {
-                {
-                    (
-                        self.functions[def_index].0 as *const _,
-                        VMFunctionEnvironment {
-                            vmctx: self.vmctx_ptr(),
-                        },
-                    )
-                } else {
-                    let import = self.imported_function(*index);
-                    (import.body, import.environment)
-                };
-                let call_trampoline = Some(self.function_call_trampolines[*sig_index]);
-                let signature = self.module.signatures[*sig_index].clone();
-                ExportFunction {
-                    address,
-                    // Any function received is already static at this point as:
-                    // 1. All locally defined functions in the Wasm have a static signature.
-                    // 2. All the imported functions are already static (because
-                    //    they point to the trampolines rather than the dynamic addresses).
-                    kind: VMFunctionKind::Static,
-                    signature,
-                    vmctx,
-                    call_trampoline,
-                }
-                .into()
-            }
-            ExportIndex::Table(index) => {
-                let from = if let Some(def_index) = self.module.local_table_index(*index) {
-                    self.tables[def_index].clone()
-                } else {
-                    let import = self.imported_table(*index);
-                    import.from.clone()
-                };
-                ExportTable { from }.into()
-            }
-            ExportIndex::Memory(index) => {
-                let from = if let Some(def_index) = self.module.local_memory_index(*index) {
-                    self.memories[def_index].clone()
-                } else {
-                    let import = self.imported_memory(*index);
-                    import.from.clone()
-                };
-                ExportMemory { from }.into()
-            }
-            ExportIndex::Global(index) => {
-                let from = {
-                    if let Some(def_index) = self.module.local_global_index(*index) {
-                        self.globals[def_index].clone()
-                    } else {
-                        let import = self.imported_global(*index);
-                        import.from.clone()
-                    }
-                };
-                ExportGlobal { from }.into()
-            }
-        }
-    }
-
-    /// Return an iterator over the exports of this instance.
-    ///
-    /// Specifically, it provides access to the key-value pairs, where the keys
-    /// are export names, and the values are export declarations which can be
-    /// resolved `lookup_by_declaration`.
-    pub fn exports(&self) -> indexmap::map::Iter<String, ExportIndex> {
-        self.module.exports.iter()
-    }
-
```

Comment:
Moved inside `InstanceHandle`, so that we can inject `InstanceAllocator` clones inside the exports.